### PR TITLE
[vulkan-memory-allocator] Update port to current ToT

### PR DESCRIPTION
**Pull request by juan-lunarg**
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [x] SHA512s are updated for each updated download
- [x] The "supports" clause reflects platforms that may be fixed by this new version
- [x] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

Contains fixes for CMake installation and removes gcc13 patch which has been fixed for a while now.

See relevant PRs:

- https://github.com/GPUOpen-LibrariesAndSDKs/VulkanMemoryAllocator/pull/350
- https://github.com/GPUOpen-LibrariesAndSDKs/VulkanMemoryAllocator/pull/326

### CODE REVIEW

File: ports/vulkan-memory-allocator/vcpkg.json [hunk 1]
```diff
@@ -1,9 +1,19 @@
 {
   "name": "vulkan-memory-allocator",
   "version": "3.0.1",
```
Comment (JonLiu1993) [inline ports/vulkan-memory-allocator/vcpkg.json:3]:
```suggestion
  "version-date": "2023-.7-12",
```

File: ports/vulkan-memory-allocator/vcpkg.json [hunk 2]
```diff
@@ -1,9 +1,19 @@
...
-  "port-version": 2,
+  "port-version": 3,
```
Comment (JonLiu1993) [inline ports/vulkan-memory-allocator/vcpkg.json:4]:
```suggestion
```

**Comment by JonLiu1993** [comment]
@juan-lunarg, Thanks for your PR, when I tested the usage, I get this error:

CMakeLists:
````
cmake_minimum_required(VERSION 3.16)

project(test)

find_package(VulkanMemoryAllocator CONFIG REQUIRED)

add_executable(main usageTest.cpp)

target_link_libraries(main PRIVATE GPUOpen::VulkanMemoryAllocator)
````
usageTest.cpp:
````
#include <stdio.h>
#include <vk_mem_alloc.h>

int main()
{
    printf("%s\n", "HELLO WORLD");
    return 0;
}
````
config:
````
1> CMake generation started for configuration: 'x64-Debug'.
1> Command line: "C:\WINDOWS\system32\cmd.exe" /c "%SYSTEMROOT%\System32\chcp.com 65001 >NUL && "C:\PROGRAM FILES\MICROSOFT VISUAL STUDIO\2022\COMMUNITY\COMMON7\IDE\COMMONEXTENSIONS\MICROSOFT\CMAKE\CMake\bin\cmake.exe"  -G "Visual Studio 17 2022" -A x64  -DCMAKE_CONFIGURATION_TYPES:STRING="Debug" -DCMAKE_INSTALL_PREFIX:PATH="C:\Users\test\source\repos\usageTest\out\install\x64-Debug" -DCMAKE_TOOLCHAIN_FILE=F:/Feature-test/vulkan/vcpkg/scripts/buildsystems/vcpkg.cmake "C:\Users\test\source\repos\usageTest" 2>&1"
1> Working directory: C:\Users\test\source\repos\usageTest\out\build\x64-Debug
1> [CMake] -- Selecting Windows SDK version 10.0.22621.0 to target Windows 10.0.19045.
1> [CMake] -- The C compiler identification is MSVC 19.36.32535.0
1> [CMake] -- The CXX compiler identification is MSVC 19.36.32535.0
1> [CMake] -- Detecting C compiler ABI info
1> [CMake] -- Detecting C compiler ABI info - done
1> [CMake] -- Check for working C compiler: C:/Program Files/Microsoft Visual Studio/2022/Community/VC/Tools/MSVC/14.36.32532/bin/Hostx64/x64/cl.exe - skipped
1> [CMake] -- Detecting C compile features
1> [CMake] -- Detecting C compile features - done
1> [CMake] -- Detecting CXX compiler ABI info
1> [CMake] -- Detecting CXX compiler ABI info - done
1> [CMake] -- Check for working CXX compiler: C:/Program Files/Microsoft Visual Studio/2022/Community/VC/Tools/MSVC/14.36.32532/bin/Hostx64/x64/cl.exe - skipped
1> [CMake] -- Detecting CXX compile features
1> [CMake] -- Detecting CXX compile features - done
1> [CMake] -- Configuring done (8.1s)
1> [CMake] -- Generating done (0.0s)
1> [CMake] -- Build files have been written to: C:/Users/test/source/repos/usageTest/out/build/x64-debug
1> Extracted CMake variables.
1> Extracted source files and headers.
1> Extracted code model.
1> Extracted toolchain configurations.
1> Extracted includes paths.
1> CMake generation finished.

````
build:
````
>------ Build All started: Project: usageTest, Configuration: x64-Debug ------
  MSBuild version 17.6.3+07e294721 for .NET Framework
  
    Checking Build System
    Building Custom Rule C:/Users/test/source/repos/usageTest/usageTest/CMakeLists.txt
    usageTest.cpp
F:\Feature-test\vulkan\vcpkg\installed\x64-windows\include\vk_mem_alloc.h(129,10): fatal error C1083: Cannot open include file: 'vulkan/vulkan.h': No such file or directory 

Build All failed.
````

When I add `find_package(Vulkan REQUIRED)` and `target_link_libraries(main PRIVATE Vulkan::Vulkan)` to CMakeLists, it build successfully.
````
cmake_minimum_required(VERSION 3.16)

project(test)

find_package(VulkanMemoryAllocator CONFIG REQUIRED)
find_package(Vulkan REQUIRED)

add_executable(main usageTest.cpp)

target_link_libraries(main PRIVATE GPUOpen::VulkanMemoryAllocator)
target_link_libraries(main PRIVATE Vulkan::Vulkan)
````
````
>------ Build All started: Project: usageTest, Configuration: x64-Debug ------
  MSBuild version 17.6.3+07e294721 for .NET Framework
  
    Checking Build System
    Building Custom Rule C:/Users/test/source/repos/usageTest/usageTest/CMakeLists.txt
    usageTest.cpp
    main.vcxproj -> C:\Users\test\source\repos\usageTest\out\build\x64-debug\usageTest\Debug\main.exe
    Building Custom Rule C:/Users/test/source/repos/usageTest/CMakeLists.txt

Build All succeeded.
````
I guess we need add `vulkan` dependence for `vulkan-memory-allocator`?

**Comment by JonLiu1993** [comment]
Note: I will be converting your PR to draft status. When you respond, please revert to "ready for review". That way, I can be aware that you've responded since you can't modify the tags.

**Comment by juan-lunarg** [comment]
> I guess we need add vulkan dependence for vulkan-memory-allocator?

I've added `vulkan` to as a dependency.

File: ports/vulkan-memory-allocator/vcpkg.json [hunk 3]
```diff
@@ -1,9 +1,20 @@
...
+  "dependencies": [
+    {
+      "name": "vcpkg-cmake",
+      "host": true
+    },
+    {
+      "name": "vcpkg-cmake-config",
+      "host": true
+    },
+    "vulkan"
```
Comment (juan-lunarg) [inline ports/vulkan-memory-allocator/vcpkg.json:18]: Added as requested by @JonLiu1993

**Comment by JonLiu1993** [comment]
@juan-lunarg, I tested again but get the same error:
````
>------ Build All started: Project: usageTest, Configuration: x64-Debug ------
  MSBuild version 17.6.3+07e294721 for .NET Framework
  
    Checking Build System
    Building Custom Rule C:/Users/test/source/repos/usageTest/usageTest/CMakeLists.txt
    usageTest.cpp
F:\Feature-test\vulkan\vcpkg\installed\x64-windows\include\vk_mem_alloc.h(129,10): fatal error C1083: Cannot open include file: 'vulkan/vulkan.h': No such file or directory 

Build All failed.
````

**Comment by juan-lunarg** [comment]
@JonLiu1993 I'm currently confused on how you want me to proceed.

`VulkanMemoryAllocator` has a dependency on `vulkan.h`.

For historical reasons there are lots of ways to find `vulkan.h`.

Due to the VulkanSDK not providing CMake package configuration support CMake has developed a module to help find components in the VulkanSDK.

https://cmake.org/cmake/help/latest/module/FindVulkan.html#module:FindVulkan

1.) find_package(Vulkan) -> Vulkan::Vulkan

```
find_package(Vulkan)
target_link_libraries(main PRIVATE Vulkan::Vulkan) <- This will add the vulkan headers include directory and link the vulkan library
```

2.) find_package(Vulkan) -> Vulkan::Headers

```
# Requires CMake 3.21 support
find_package(Vulkan)
target_link_libraries(main PRIVATE Vulkan::Headers) <- This will only add the vulkan headers include directory
```

3.) find_package(VulkanHeaders CONFIG) -> Vulkan::Headers

I added this in 2022 to the Vulkan::Headers repository.

```
find_package(VulkanHeaders CONFIG)
target_link_libraries(main PRIVATE Vulkan::Headers) <- This will only add the vulkan headers include directory
```

4.) add_subdirectory

```
add_subdirectory(Vulkan-Headers)
target_link_libraries(main PRIVATE Vulkan::Headers) <- This will only add the vulkan headers include directory
```

As a result the find_package for VulkanMemoryAllocator doesn't try to find the vulkan headers. It assumes you have added it to your project in whatever manner the author saw fit.

```
# Use any of the aforementioned methods of finding the vulkan::headers
# Then find the vulkan memory allocator

find_package(VulkanMemoryAllocator CONFIG REQUIRED)
target_link_libraries(main PRIVATE GPUOpen::VulkanMemoryAllocator)
```

File: ports/vulkan-memory-allocator/unofficial-vulkan-memory-allocator-config.cmake
Comment (juan-lunarg) [inline ports/vulkan-memory-allocator/unofficial-vulkan-memory-allocator-config.cmake:1]: What's also worth noting is that the current port has the same issue you are pointing out in your comment here:
https://github.com/microsoft/vcpkg/pull/32513#issuecomment-1631943478

File: ports/vulkan-memory-allocator/vcpkg.json [hunk 4]
```diff
@@ -1,9 +1,19 @@
...
   "name": "vulkan-memory-allocator",
   "version": "3.0.1",
-  "port-version": 2,
+  "port-version": 3,
   "description": "Easy to integrate Vulkan memory allocation library from GPUOpen",
   "homepage": "https://github.com/GPUOpen-LibrariesAndSDKs/VulkanMemoryAllocator",
   "license": "MIT",
-  "supports": "!uwp & !xbox"
+  "supports": "!uwp & !xbox",
+  "dependencies": [
```
Comment (juan-lunarg) [inline ports/vulkan-memory-allocator/vcpkg.json:9]: To summarize this port has a dependency on either the `vulkan` port or the `vulkan-headers` port.

And I'm not sure how to express this to users in the `usage` file.

**Comment by JonLiu1993** [comment]
@juan-lunarg, I investigated this error, and it has nothing to do with your PR. The usage you provided is correct. The reason for the error is that the `.cmake` file and the `.h` file are not provided after `vulkan` installation. I will allow this PR to be merged, sorry to you trouble.

**Review by JonLiu1993 — DISMISSED** [review]

**Comment by juan-lunarg** [comment]
> The reason for the error is that the .cmake file and the .h file are not provided after vulkan installation. I will allow this PR to be merged, sorry to you trouble.

No worries.

File: ports/vulkan-memory-allocator/usage [hunk 5]
```diff
@@ -0,0 +1,4 @@
+VulkanMemoryAllocator provides official find_package support:
```
Comment (BillyONeal) [inline ports/vulkan-memory-allocator/usage:1]: Would you accept:

```
VulkanMemoryAllocator provides CMake targets. These targets require that Vulcan headers are available while building the target, either by CMake's official find module, the vulcan-headers port, or by add_subdirectory. For example:

    find_package(Vulkan) # https://cmake.org/cmake/help/latest/module/FindVulkan.html
    find_package(VulkanMemoryAllocator CONFIG REQUIRED)
    target_link_libraries(main PRIVATE Vulkan::Headers GPUOpen::VulkanMemoryAllocator)

or:

    find_package(VulkanHeaders CONFIG) # From the port vulcan-headers
    find_package(VulkanMemoryAllocator CONFIG REQUIRED)
    target_link_libraries(main PRIVATE Vulkan::Headers GPUOpen::VulkanMemoryAllocator)
```

**Comment by BillyONeal** [comment]
Thanks for the update! I really like these changes, just want your thoughts at including more of your awesome writeup ( https://github.com/microsoft/vcpkg/pull/32513#issuecomment-1634663791 ) in the usage so that we and users aren't confused in the future.

File: ports/vulkan-memory-allocator/usage (hunk 5 above)
Comment (juan-lunarg) [reply ports/vulkan-memory-allocator/usage:1]: Looks good to me 👍🏾 , I'll fix up my PR.

Comment (juan-lunarg) [reply ports/vulkan-memory-allocator/usage]: I've updated the usage let me know what you think.

Comment (BillyONeal) [reply ports/vulkan-memory-allocator/usage]: I pushed a couple of nitpick fixes and now LGTM! Thanks!

**Review by BillyONeal — APPROVED** [review]

**Comment by BillyONeal** [comment]
Will merge as soon as I notice PR build comes back (probably later this evening). Thanks again!

**Pull request merged**